### PR TITLE
ranger, stats: remove Align method of IndexRange.

### PR DESCRIPTION
**Pull request by winoros**
We don't need this method.
PTAL @hanfei1991 @coocood @lamxTyler

### CODE REVIEW

File: statistics/statistics_test.go [hunk 1]
```diff
@@ -631,12 +631,12 @@ func (s *testStatisticsSuite) TestIndexRanges(c *C) {
 	ran[0].HighVal[0] = types.NewIntDatum(2000)
 	count, err = tbl.GetRowCountByIndexRanges(sc, 0, ran)
 	c.Assert(err, IsNil)
-	c.Assert(int(count), Equals, 999)
+	c.Assert(int(count), Equals, 1000)
```
Comment (winoros) [inline statistics/statistics_test.go]: the real count is 1000.

File: statistics/statistics_test.go [hunk 2]
```diff
@@ -631,12 +631,12 @@ func (s *testStatisticsSuite) TestIndexRanges(c *C) {
...
 	ran[0].LowVal[0] = types.NewIntDatum(1001)
 	ran[0].HighVal[0] = types.NewIntDatum(1990)
 	count, err = tbl.GetRowCountByIndexRanges(sc, 0, ran)
 	c.Assert(err, IsNil)
-	c.Assert(int(count), Equals, 988)
+	c.Assert(int(count), Equals, 989)
```
Comment (winoros) [inline statistics/statistics_test.go]: the real count is 990.

File: statistics/histogram.go [hunk 3]
```diff
@@ -581,7 +582,7 @@ func (idx *Index) equalRowCount(sc *stmtctx.StatementContext, b []byte) (float64
 func (idx *Index) getRowCount(sc *stmtctx.StatementContext, indexRanges []*ranger.IndexRange) (float64, error) {
 	totalCount := float64(0)
 	for _, indexRange := range indexRanges {
-		indexRange.Align(len(idx.Info.Columns))
+		isPoint := len(indexRange.LowVal) == len(indexRange.HighVal) && len(indexRange.LowVal) == len(idx.Info.Columns)
```
Comment (lamxTyler) [inline statistics/histogram.go]: Is `isPoint` a little miss leading? Maybe `isFullRange` is better. Also, you can put this declaration near where it is used.

**Pull request merged**